x-pack/plugins/monitoring/server/kibana_monitoring/bulk_uploader.js

```diff
@@ -57,6 +57,10 @@ export class BulkUploader {
 
     this._callClusterWithInternalUser = callClusterFactory(server).getCallClusterInternal();
 
+    const repository = server.savedObjects.getSavedObjectsRepository(
+      this._callClusterWithInternalUser
+    );
+    this._savedObjectsClient = new server.savedObjects.SavedObjectsClient(repository);
```

Comment:
@kobelb do you think it's okay to move forward with this for now?

Reply:
Using the hard-coded default SavedObjectsClient constructor here gets us little benefit, I'd suggest using the Repository directly. The Repository is completely unsecured, so you don't want to use it in the context of end-user requests, but if I understand correctly the Kibana server itself is collecting the stats periodically, so using the repository directly is recommended.

Reply:
I spoke with @kobelb and found out this comment would only apply to internal collection, but as its unsecured, it's not what we'd want to do when the collector fetch methods are called to expose stats in an API, which is the future plan for ingesting monitoring stats (using Metricbeat).

This PR is stalled until we come up with a plan for a secure way to fetch monitoring stats "live" from collectors for the stats API.